tests/test_metadata_cli.py

```diff
@@ -0,0 +1,234 @@
+import sys
+import json
+
+import responses
+from mock import patch
+from pytest import mark
+
+if sys.version_info.major == 3:
+    builtin_module_name = 'builtins'
+else:
+    builtin_module_name = '__builtin__'
+
+
+# GET
+@responses.activate
+def test_metadata_get(cli_run):
+    responses.add(
+        responses.GET,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid',
+        json={'metadata': {
+            "general_title": "test"
+        }},
+        status=200)
+
+    res = cli_run("metadata get -p some-pid")
+
+    assert responses.calls[0].request.headers[
+        'Accept'] == 'application/basic+json'
+    assert res.exit_code == 0
+    assert json.loads(res.stripped_output) == {"general_title": "test"}
+
+
+@responses.activate
+def test_metadata_get_field(cli_run):
+    responses.add(
+        responses.GET,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid',
+        json={'metadata': {
+            "general_title": "test"
+        }},
+        status=200)
+
+    res = cli_run("metadata get -p some-pid -f general_title")
+
+    assert responses.calls[0].request.headers[
+        'Accept'] == 'application/basic+json'
+    assert res.exit_code == 0
+    assert res.stripped_output == '"test"'
+
+
+@responses.activate
+def test_metadata_get_pid_not_exists(cli_run):
+    responses.add(
+        responses.GET,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid',
+        json={
+            'status': 404,
+            'message': 'PID does not exist.'
+        },
+        status=404)
+
+    res = cli_run("metadata get -p some-pid")
+
+    assert res.exit_code == 1
+    assert res.stripped_output == 'PID does not exist.'
+
+
+def test_metadata_get_no_pid_given(cli_run):
+    res = cli_run("metadata get")
+
+    assert res.exit_code == 2
+    assert "Error: Missing option '--pid' / '-p'." in res.output
+
+
+# UPDATE
+@mark.skip
+@responses.activate
+def test_metadata_update_with_text_field(cli_run):
+    responses.add(
+        responses.PATCH,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid',
+        json={'metadata': {
+            "general_title": "new-test"
+        }},
+        status=200)
+
+    res = cli_run(
+        "metadata update -p some-pid --field general_title -j new-test")
+
+    assert responses.calls[0].request.headers[
+        'Accept'] == 'application/basic+json'
+    assert responses.calls[0].request.headers[
+        'Content-Type'] == 'application/json-patch+json'
+
+    assert json.loads(responses.calls[0].request.body) == [{
+        'op': 'replace',
+        'path': '/general_title',
+        'value': 'new-test'
+    }]
+
+    assert res.exit_code == 0
+    assert json.loads(res.stripped_output) == {
+        'metadata': {
+            "general_title": "new-test"
+        }
+    }
+
+
+@responses.activate
+def test_metadata_update_with_json_field(cli_run):
```

Comment:
we need one more test when updating whole metadata (without specifying a field)
*important as we use different method in this case (update instead of patch)*